projects/core/src/cart/store/reducers/index.ts

```diff
@@ -28,10 +30,17 @@ export function clearCartState(
   reducer: ActionReducer<any>
 ): ActionReducer<any> {
   return function(state, action) {
+    let newAction = action;
     if (action.type === LOGOUT || action.type === PLACE_ORDER_SUCCESS) {
+      console.log('yaba');
       state = undefined;
+    } else if (
+      action.type === LANGUAGE_CHANGE ||
+      action.type === CURRENCY_CHANGE
+    ) {
+      newAction = new fromActions.ResetCartDetails();
```

Comment:
If possible, it would be good to dispatch a new action in the cart.effect. Optionally, please create a separate ticket for it.